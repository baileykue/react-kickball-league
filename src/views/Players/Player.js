import { useState, useEffect } from 'react';
import { getPlayerById } from '../../services/players';
import PlayerDetail from '../../components/PlayerList/PlayerDetail';

export default function Player(props) {
  const id = props.match.params.id;
  const [player, setPlayer] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPlayerById(id);
      setPlayer(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h3>Loading team...</h3>;

  return (
    <>
      <PlayerDetail player={player} showDetail />
    </>
  );
}
