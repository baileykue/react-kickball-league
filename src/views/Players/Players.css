.players {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 25px;
}

.detail {
  display: flex;
  flex-direction: column;
  align-items: center;
}
