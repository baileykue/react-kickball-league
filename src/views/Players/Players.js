import React, { useState, useEffect } from 'react';
import { fetchPlayers } from '../../services/players';
import PlayerList from '../../components/PlayerList/PlayerList';

export default function Players() {
  const [players, setPlayers] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayers();
      setPlayers(data);
    };
    fetchData();
  }, []);

  return (
    <div>
      <h1>This is the players page</h1>
      {players.map((player) => (
        <PlayerList key={player.id} {...player} />
      ))}
    </div>
  );
}
