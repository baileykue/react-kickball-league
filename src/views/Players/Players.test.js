import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Players from './Players';

test('should render all players', async () => {
  const { container } = render(
    <MemoryRouter>
      <Players />
    </MemoryRouter>
  );
  await screen.findByText('Dwayne Elizondo Mountain Dew Herbert Camacho');
  expect(container).toMatchSnapshot();
});
