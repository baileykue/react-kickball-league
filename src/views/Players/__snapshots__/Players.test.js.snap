// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all players 1`] = `
<div>
  <div
    class="players"
  >
    <h1>
      These are the players
    </h1>
    <div>
      <a
        href="/players/2"
      >
        Hans Olo
      </a>
    </div>
    <div>
      <a
        href="/players/3"
      >
        Ron Long
      </a>
    </div>
    <div>
      <a
        href="/players/4"
      >
        Shorty Biggum
      </a>
    </div>
    <div>
      <a
        href="/players/5"
      >
        Gary Coolman
      </a>
    </div>
    <div>
      <a
        href="/players/6"
      >
        Terry Cruise
      </a>
    </div>
    <div>
      <a
        href="/players/7"
      >
        Bud E. Guy
      </a>
    </div>
    <div>
      <a
        href="/players/8"
      >
        Buster Rimes
      </a>
    </div>
    <div>
      <a
        href="/players/9"
      >
        Deb E. Downer
      </a>
    </div>
    <div>
      <a
        href="/players/10"
      >
        Jayne Phonda
      </a>
    </div>
    <div>
      <a
        href="/players/11"
      >
        Kara Whollot
      </a>
    </div>
    <div>
      <a
        href="/players/12"
      >
        Betty Grey
      </a>
    </div>
    <div>
      <a
        href="/players/13"
      >
        Dwayne Elizondo Mountain Dew Herbert Camacho
      </a>
    </div>
    <div>
      <a
        href="/players/14"
      >
        Joe Bauers
      </a>
    </div>
    <div>
      <a
        href="/players/15"
      >
        Frito P.
      </a>
    </div>
    <div>
      <a
        href="/players/1"
      >
        Bennie Jetts
      </a>
    </div>
  </div>
</div>
`;
